ipatests/test_webui/test_selfservice.py

```diff
@@ -49,4 +59,230 @@ def test_crud(self):
+        self.navigate_to_record(data_selfservice.DATA1['pkey'])
+        self.check_option('attrs', 'businesscategory')
+        self.facet_button_click('save')
+        self.assert_field_validation(FIELD_REQ, field='attrs')
+
+    @screenshot
+    def test_del_multiple_permission(self):
+        """
+        Try to delete multiple self service permission
+        """
+        self.init_app()
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA1)
+        self.add_record(data_selfservice.ENTITY, data_selfservice.DATA)
+        self.delete_record([data_selfservice.DATA1['pkey'],
+                            data_selfservice.DATA['pkey']])
+
+    @screenshot
+    def test_permission_using_enter_key(self):
+        """
+        Try to add/delete persmission using enter key
+        """
+        # try to add using enter key
+        self.init_app()
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.wait_for_request()
+        self.facet_button_click('add')
+        self.fill_textbox('aciname', data_selfservice.DATA1['pkey'])
+        self.check_option('attrs', 'audio')
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait()
+        self.assert_notification(assert_text=SERVICE_ADDED)
+        self.assert_record(data_selfservice.DATA1['pkey'])
+        self.close_notifications()
+
+        # try to delete using enter key
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.select_record(data_selfservice.DATA1['pkey'])
+        self.facet_button_click('remove')
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait()
+        self.assert_notification(assert_text='1 item(s) deleted')
+        self.close_notifications()
+
+    @screenshot
+    def test_reset_sshkey_permsission(self):
+        """
+        Try to delete sshkey after altering sshkey permission
+        """
+        pwd = self.config.get('ipa_password')
+
+        self.init_app()
+        self.add_record(user.ENTITY, user.DATA, navigate=False)
+        reset_passwd(self, user.PKEY, pwd)
+        self.logout()
+        self.login(user.PKEY, password=pwd, new_password=pwd)
+        self.add_sshkey_to_record(user.SSH_RSA, user.PKEY, navigate=True)
+        self.assert_num_ssh_keys(1)
+        self.close_notifications()
+        self.logout()
+
+        self.init_app()
+        self.navigate_to_entity(data_selfservice.ENTITY)
+        self.navigate_to_record('Users can manage their own SSH public keys')
+        self.check_option('attrs', 'carlicense')
+        self.check_option('attrs', 'ipasshpubkey')
+        self.facet_button_click('save')
+        self.close_notifications()
+        self.logout()
+
+        # check if delete button is visible for ssh key
+        self.login(user.PKEY, password=pwd)
+        self.assert_disabled('[name="ipasshpubkey"]', negative=True)
```

Comment:
This test is not right. `self.assert_disabled(.., negative=True)` basically test if it is enabled. This assert will pass in every situation regardless of previous steps. The test needs to check if a user can add or remove SSH Keys. 

There are two ways how to do it, maybe combination is correct:

- SSH Keys widget doesn't have, "add" and "delete" buttons visible
- After clicking on "Show/Set key", the textarea in the dialog is editable.